In [1]:
import numpy as np
import scipy.stats as stats


In [2]:
observed_data = np.array([
    [50,70],
    [80,100],
    [60,90],
    [30,50],
    [20,50]
])

In [4]:
chi2_stat, p_val, dof, expected = stats.chi2_contingency(observed_data)

In [5]:
alpha = 0.05
critical_value = stats.chi2.ppf(1-alpha, dof)

In [6]:
print(f"Chi-Square Statistics: {chi2_stat:.2f}")
print(f"Critical Value at alpha={alpha}: {critical_value:.2f}")
print(f"P-value: {p_val:.3f}")
print(f"Degrees of Freedom: {dof}")
print("\nExpected Frequencies (if there's no association between device type and satisfication):")
print(expected)

Chi-Square Statistics: 5.64
Critical Value at alpha=0.05: 9.49
P-value: 0.228
Degrees of Freedom: 4

Expected Frequencies (if there's no association between device type and satisfication):
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [7]:
if chi2_stat > critical_value:
  print("\nConclusion: We reject the null hypothesis.")
  print("There is a siginificant association between the type of device and customer satisfication.")
else:
  print("\nConclusion: We fail to reject the null hypothesis.")
  print("There is no significant association between the type of device and customer satisfication.")


Conclusion: We fail to reject the null hypothesis.
There is no significant association between the type of device and customer satisfication.


**Analysis**

1. **Chi-Square Statistic**: The calculated Chi-Square statistic is approximately
5.64.

2. **Critical Value**: The critical value for 4 degrees of freedom at a significance level of α=0.05 is approximately 9.49.

3. **p-value**: The p-value associated with the Chi-Square statistic is approximately 0.228.


Since the Chi-Square statistic 5.64 is less than the critical value 9.49,and the p-value 0.228 is greater than 0.05, we do not reject the null hypothesis.




**Conclusion**

There is no significant association between the type of smart home device (Smart Thermostats vs. Smart Lights) and customer satisfaction levels.